packages/streamline/src/components/checkbox/checkbox.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import { fireEvent } from '@testing-library/react-native';
+import { renderWithProvider } from '../../testing/render-with-provider';
+import { Checkbox } from './checkbox';
+
+describe('Checkbox', () => {
+  it('should render successfully', () => {
+    const { container } = renderWithProvider(<Checkbox value={true} />);
+    expect(container).toBeTruthy();
+  });
+
+  it('should trigger mockOnChange', () => {
+    const mockOnChange = jest.fn();
+    const { getByTestId } = renderWithProvider(
+      <Checkbox value={true} onChange={mockOnChange} testID="Checkbox" />
+    );
+    const checkbox = getByTestId('Checkbox');
+    fireEvent.press(checkbox);
+    expect(mockOnChange).toHaveBeenCalledTimes(1);
```

Comment:
```suggestion
    expect(mockOnChange).toHaveBeenNthCalledWith(1, false);
```

Reply:
To improve testing you could also check that the checked icon is visible, then not visible anymore WDYT?